src/pipelinedefinition.cpp

```diff
@@ -384,4 +384,63 @@ Status PipelineDefinition::getInputsInfo(tensor_map_t& inputsInfo, ModelManager&
     return StatusCode::OK;
 }
 
+Status PipelineDefinition::getOutputsInfo(tensor_map_t& outputsInfo, const ModelManager& manager) const {
+    // Assumptions: this can only be called on available pipeline definition.
+    // Add check if available when pipeline status will be implemented.
```

Comment:
Side comment:
probably we should throw when PipelineStatus is not in AVAILABLE/AVAILABLE_LIKE state. However we do not have status yet.

Reply:
Yes, when pipeline status is added, we can throw at the beginning of this function if it is not in available state. For the period of getInputsInfo/getOutputsInfo we shall keep unload guard and keep it available.